table/block_based/block_based_table_iterator.cc

```diff
@@ -18,139 +18,149 @@ void BlockBasedTableIterator::Seek(const Slice& target) {
 
 void BlockBasedTableIterator::SeekImpl(const Slice* target,
                                        bool async_prefetch) {
-  ResetBlockCacheLookupVar();
   bool is_first_pass = !async_read_in_progress_;
```

Comment:
Looks like almost all of this function is for `is_first_pass == true`. Maybe the second pass can be separated out into another function to make it easier to read.